src/main/resources/assets/computercraft/lua/rom/modules/cctpm.lua

```diff
@@ -0,0 +1,248 @@
+local cctpm = {}
+
+cctpm_cache = {}
+
+local function downloadFile( sURL, sPath )
+    local response, err = http.get( sURL )
+    if err then
+        return false, err
+    end
+    local handle = fs.open( sPath, "w" )
+    handle.write( response.readAll() )
+    response.close()
+    handle.close()
+    return true
+end
+
+local function addRepo( sRepo, sName, tInstalledList )
+    local tRepo = textutils.unserialise( sRepo )
+    if type( tRepo ) ~= "table" then
+        return false, "Can't Read Repo \""..sName..'"'
+    end
+    for a, b in pairs( tRepo) do 
+        cctpm_cache.packages[a] = b
+        if tInstalledList[a] then
+            cctpm_cache.packages[a]["installed"] = true
+        else
+            cctpm_cache.packages[a]["installed"] = false
+        end 
+    end
+    return true
+end
+
+function cctpm.readRepos()
+    cctpm_cache.packages = {}
+    local sDirectory = fs.combine( settings.get( "cctpm.directory", "/var/cctpm" ), "" )
+    local tInstalledList = {}
+    if fs.exists( sDirectory.."/installed.txt" ) then
+        for sLine in io.lines( sDirectory.."/installed.txt" ) do
+            local sPackageName = sLine:match("([^,]+);([^,]+)")
+            tInstalledList[sPackageName] = true
+        end
+    end
+    local sRepoList = settings.get( "cctpm.repolist", "" )
+        for sRepo in sRepoList:gmatch( sRepoList, "[^;]+") do
+            local sType, sMeta = sRepo:match("([^,]+)|([^,]+)")
+            if sType == "repo" then
+                sName, sURL = sMeta:match("([^,]+):([^,]+)")
+                local response, err = http.get( sURL )
+                if err then
+                    return false, err
+                end
+                local ok, err = addRepo( response.readAll() , sName, tInstalledList )
+                response.close()
+                if not ok then
+                    return false, err
+                end
+            elseif sType == "repolist" then
+                local ok, err = downloadFile( sMeta, "/tmp/repolist.txt" )
```

Comment:
Why are you not being consistent with your use of Dan's hungarian notation? (`sType`, `err`)